hack/lib/update-codegen.sh

```diff
@@ -98,33 +102,40 @@ require (
     k8s.io/api ${KUBE_MODULE_VERSION}
     k8s.io/apimachinery ${KUBE_MODULE_VERSION}
     k8s.io/client-go ${KUBE_MODULE_VERSION}
-    k8s.io/apimachinery ${KUBE_MODULE_VERSION}
+    ${BASE_PKG}/generated/${KUBE_MINOR_VERSION}/apis v0.0.0
 )
 
 replace ${BASE_PKG}/generated/${KUBE_MINOR_VERSION}/apis => ../apis
 EOF
 
+# Generate a go.sum without changing the go.mod by running go mod download.
+echo "running go mod download in ${OUTPUT_DIR}/client/go.mod to generate a go.sum file..."
+(cd "${OUTPUT_DIR}/client" && go mod download all 2>&1 | sed "s|^|go-mod-download > |")
```

Comment:
Overall I see two fixes in this script:

1. Add an explicit `require` that is replaced in the `go.mod`
2. Make sure a `go.sum` file exists

Was that all we needed to do to fix our codegen?

Reply:
Pretty much, yes. Although there's something subtle about the making sure that a valid go.sum exists that I don't fully understand. Using `go mod graph` also generates the same valid go.sum, but that wasn't enough to make codegen work. Something about using `go mod download` made it work.